### PR TITLE
PF-774 normalize dependencies and use stairway with fixed dependencies

### PR DESCRIPTION
**Pull request by ddietterich**
The last of the library PRs that I hope align the version dependencies.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,54 +28,67 @@ repositories {
 
 group 'bio.terra'
 version '0.0.44-SNAPSHOT'
```
Comment (jaycarlton) [inline build.gradle:30]: do you not need to bump this too?

File: build.gradle [hunk 2]
```diff
@@ -4,11 +4,15 @@ plugins {
     id 'java-library'
     id 'maven-publish'
 
-    id 'com.diffplug.spotless' version '5.10.2'
-    id 'com.github.ben-manes.versions' version '0.33.0'
+    id 'com.diffplug.spotless' version '5.12.4'
```
Comment (jaycarlton) [inline build.gradle:7]: If there are new rules or guidelines (such as avoiding `+`), then we should add them to the repo documentation.

File: build.gradle [hunk 3]
```diff
@@ -24,54 +28,67 @@ repositories {
...
-    implementation group: 'com.google.cloud', name: 'google-cloud-core', version: '1.94.+'
-    implementation group: 'com.google.code.findbugs', name: 'annotations', version: '3.+'
-    implementation group: 'com.google.apis', name: 'google-api-services-logging', version: 'v2+'
+
+
+    // Google dependencies - make sure we are using -jre and not android and use common bom
+    constraints {
+        implementation 'com.google.guava:guava:30.1.1-jre' // "-jre" for Java 8 or higher
+    }
+    implementation platform('com.google.cloud:libraries-bom:20.2.0')
```
Comment (jaycarlton) [inline build.gradle:43]: Can you alphabetize these please?

File: build.gradle [hunk 4]
```diff
@@ -24,54 +28,67 @@ repositories {
...
+    implementation group: 'bio.terra', name: 'stairway', version: '0.0.51-SNAPSHOT'
+    implementation group: 'org.postgresql', name: 'postgresql', version: '42.2.18'
+
+    implementation group: 'ch.qos.logback.contrib', name: 'logback-json-classic', version: '0.1.5'
+    implementation group: 'ch.qos.logback.contrib', name: 'logback-jackson', version: '0.1.5'
+    implementation group: 'com.google.cloud', name: 'google-cloud-core'
+    implementation group: 'com.google.code.findbugs', name: 'annotations', version: '3.0.1'
+    implementation group: 'com.google.apis', name: 'google-api-services-logging', version: 'v2-rev20210507-1.31.0'
     implementation group: 'io.kubernetes', name: 'client-java', version: '10.0.0'
     implementation group: 'io.opencensus', name: 'opencensus-exporter-trace-stackdriver', version: opencensus
```
Comment (jaycarlton) [inline build.gradle:54]: Is `opencensus` the same as `project.ext.opencensus` or whatever the incantation is?

File: build.gradle [hunk 5]
```diff
@@ -24,54 +28,67 @@ repositories {
...
-    implementation group: 'org.apache.commons', name: 'commons-lang3'
-    implementation group: 'org.apache.commons', name: 'commons-pool2'
-    implementation group: 'org.hashids', name: 'hashids', version: '1.+'
-    implementation group: 'org.liquibase' , name: 'liquibase-core', version: '3.10.0'
-    implementation group: 'org.slf4j', name: 'slf4j-api', version: '1.7.+'
-    implementation group: 'org.springframework' , name: 'spring-context'
-    implementation group: 'org.springframework' , name: 'spring-core'
-    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-data-jdbc'
-    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-web'
+    implementation group: 'org.apache.commons', name: 'commons-dbcp2', version: '2.8.0'
```
Comment (jaycarlton) [inline build.gradle:56]: were these Spring Boot dependencies before? Are we moving away from that (in all the repos)? I thought it was supposed to be one of the killer features of Boot.

File: build.gradle [hunk 6]
```diff
@@ -24,54 +28,67 @@ repositories {
...
+    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.11'
+    implementation group: 'org.apache.commons', name: 'commons-pool2', version: '2.9.0'
+    implementation group: 'org.hashids', name: 'hashids', version: '1.0.3'
+    implementation group: 'org.liquibase' , name: 'liquibase-core', version: '4.2.1'
+    implementation group: 'org.slf4j', name: 'slf4j-api', version: '1.7.30'
+    implementation group: 'org.springframework' , name: 'spring-context', version: '5.3.7'
+    implementation group: 'org.springframework' , name: 'spring-core', version: '5.3.7'
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-data-jdbc', version: '2.4.5'
```
Comment (jaycarlton) [inline build.gradle:64]: Where did you get all these version numbers? I don't know what's a real change and what isn't. I suppose `gradle dependencies` is the way to get them?

File: build.gradle [hunk 7]
```diff
@@ -92,7 +109,7 @@ if (hasProperty('buildScan')) {
     }
 }
 
-apply from: "$rootDir/gradle/dependency-locking.gradle"
+//apply from: "$rootDir/gradle/dependency-locking.gradle"
```
Comment (jaycarlton) [inline build.gradle:112]: why is this commented out?

File: src/main/java/bio/terra/common/logging/RequestCacheFilter.java [hunk 8]
```diff
@@ -28,4 +29,10 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
     chain.doFilter(reqWrapper, resWrapper);
     resWrapper.copyBodyToResponse();
   }
+
+  @Override
```
Comment (jaycarlton) [inline src/main/java/bio/terra/common/logging/RequestCacheFilter.java:33]: do these overrides serve to keep the filters alive forever? Seems odd you'd need to override with empty bodies.

File: src/test/java/bio/terra/common/sam/SamRetryTest.java [hunk 9]
```diff
@@ -17,7 +17,7 @@
 @Tag("unit")
 public class SamRetryTest {
 
-  private static Logger logger = LoggerFactory.getLogger(SamRetryTest.class);
+  private static final Logger logger = LoggerFactory.getLogger(SamRetryTest.class);
```
Comment (jaycarlton) [inline src/test/java/bio/terra/common/sam/SamRetryTest.java:20]: aside: Is it better for loggers to be static? Many of the ones in wsm are not today.

File: src/test/java/bio/terra/common/tracing/TracingTest.java [hunk 10]
```diff
@@ -9,19 +9,16 @@
...
+import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
 import org.springframework.boot.test.web.client.TestRestTemplate;
 import org.springframework.http.ResponseEntity;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
 
 /** Tests the functionality of the common tracing package. */
```
Comment (jaycarlton) [inline src/test/java/bio/terra/common/tracing/TracingTest.java:21]: does this mean the jupiter test runner takes over as it's the default?

**Review by jaycarlton — APPROVED** [review]

**Review by yonghaoy — APPROVED** [review]

**Review by jmczerk — APPROVED** [review]
Thanks for doing this, Dan.  I think these changes get us in better shape around defining a platform BOM, I might take another run at this once these all land.

File: src/test/resources/application-tracing-test.properties [hunk 11]
```diff
@@ -1,3 +1,4 @@
 terra.common.tracing.samplingRate: 1.0
 # Empty list of urlPatterns to allow all URLs to be traced.
 terra.common.tracing.urlPatterns:
+terra.common.tracking.stackdriverExportEnabled: false
```
Comment (wchamber) [inline src/test/resources/application-tracing-test.properties]: typo: 'tracking' instead of 'tracing'. Hopefully works after that...

Comment (ddietterich) [reply src/test/resources/application-tracing-test.properties]: Thank  you!

**Pull request merged**